### PR TITLE
Split build/publish in CircleCI

### PR DESCRIPTION
**Pull request by SuperQ**
## Changes

* Split the test, build, and publish steps with CircleCI workflows.
* Fixup `promu` target so it can be used outside of make.
* Update git ignore list for promu use.
* Update promu crossbuild list to avoid 32-bit builds.
* Siplify publish step with workflow filter.

## Verification

Tested promu crossbuild locally.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,17 +36,28 @@ errcheck:
...
 $(GOIMPORTS):
 	@echo ">> fetching goimports"
 	@go get -u golang.org/x/tools/cmd/goimports
 
-$(PROMU):
+.PHONY: promu
+promu:
 	@echo ">> fetching promu"
-	@go get -u github.com/prometheus/promu
+	GOOS= GOARCH= go get -u github.com/prometheus/promu
```
Comment (Bplotka) [inline Makefile]: wonder why is that needed?

**Comment by Bplotka** [comment]
WiIl merge, just have one question (asked above)

File: Makefile (hunk 1 above)
Comment (SuperQ) [reply Makefile]: This allows for setting `GOOS` and `GOARCH` locally for using cross-compile with `make build`. while still keeping the `promu` dependency locally compiled.

**Comment by Bplotka** [comment]
Thanks!
